core/services/relay/evm/event_binding.go

```diff
@@ -227,32 +227,56 @@ func (e *eventBinding) getLatestValueWithFilters(
 		return err
 	}
 
-	fai := filtersAndIndices[0]
-	remainingFilters := filtersAndIndices[1:]
-
-	logs, err := e.lp.IndexedLogs(ctx, e.hash, e.address, 1, []common.Hash{fai}, confs)
+	// Create limiter and filter for the query.
+	limiter := query.NewLimitAndSort(query.CountLimit(1), query.NewSortBySequence(query.Desc))
+	filter, err := query.Where(
+		e.address.String(),
+		logpoller.NewAddressFilter(e.address),
+		logpoller.NewEventSigFilter(e.hash),
+		logpoller.NewConfirmationsFilter(confs),
+		createTopicFilters(filtersAndIndices),
+	)
 	if err != nil {
 		return wrapInternalErr(err)
 	}
 
-	// TODO Use filtered logs here BCF-3316
-	// TODO: there should be a better way to ask log poller to filter these
-	// First, you should be able to ask for as many topics to match
-	// Second, you should be able to get the latest only
-	var logToUse *logpoller.Log
-	for _, log := range logs {
-		tmp := log
-		if compareLogs(&tmp, logToUse) > 0 && matchesRemainingFilters(&tmp, remainingFilters) {
-			// copy so that it's not pointing to the changing variable
-			logToUse = &tmp
-		}
+	// Gets the latest log that matches the filter and limiter.
+	logs, err := e.lp.FilteredLogs(ctx, filter, limiter, e.contractName+"-"+e.eventName)
```

Comment:
should I add address to the `queryName` too?

Reply:
don't see why not , CC @EasterTheBunny

Reply:
Using `queryName` here might be helpful. The idea is that the string passed will show in slow query logs and help identify problems in the log file.